### PR TITLE
Update `firebase init functions` templates to 2nd gen

### PR DESCRIPTION
**Pull request by HYACCCINT**
### Description

Update `firebase init functions` JS and TS templates to 2nd gen.

### Scenarios Tested

JavaScript:
1. `firebase init functions`
2. select JavaScript
3. select `y` for ESLint
4. uncomment the generated functions code
5. `firebase deploy --only functions`

TypeScript:
1. `firebase init functions`
2. select TypeScript
3. select `y` for ESLint
4. uncomment the generated functions code
5. `firebase deploy --only functions`

### CODE REVIEW

File: templates/init/functions/javascript/index.js
Comment (jhuleatt) [inline templates/init/functions/javascript/index.js:1]: Optional idea: add some commented-out lines showing other imports, like this:

```suggestion
/**
 * Import function triggers from their respective submodules:
 *
 * const { onCall } = require("firebase-functions/v2/https");
 * const { onDocumentWritten } = require("firebase-functions/v2/firestore");
 *
 * See a full list of supported triggers at https://firebase.google.com/docs/functions
 */

const { onRequest } = require("firebase-functions/v2/https");
```

File: templates/init/functions/typescript/index.ts
Comment (taeold) [inline templates/init/functions/typescript/index.ts:2]: In TypeScript, let's prefer "import" syntax, e.g:

```
import { onRequest } from "firebase-functions/v2/https";
import * as logger from firebase-functions/logger";
```

File: templates/init/functions/typescript/index.ts
Comment (jhuleatt) [inline templates/init/functions/typescript/index.ts:5]:
```suggestion
 * import { onCall } = from "firebase-functions/v2/https";
 * import { onDocumentWritten } = from "firebase-functions/v2/firestore";
```

**Review by jhuleatt — APPROVED** [review]
Thank you @HYACCCINT!!! Please get approval from Daniel before merging

**Comment by jhuleatt** [comment]
Oh - one more thing - please add an entry to [`CHANGELOG.md`](https://github.com/firebase/firebase-tools/blob/master/CHANGELOG.md) as part of this PR

File: templates/init/functions/typescript/index.ts
Comment (jhuleatt) [inline templates/init/functions/typescript/index.ts:5]:
```suggestion
 * import { onCall } from "firebase-functions/v2/https";
 * import { onDocumentWritten } from "firebase-functions/v2/firestore";
```

File: templates/init/functions/typescript/index.ts
Comment (jhuleatt) [inline templates/init/functions/typescript/index.ts:11]:
```suggestion
import { onRequest } from "firebase-functions/v2/https";
import * as logger from "firebase-functions/logger";
```

File: templates/init/functions/typescript/index.ts
Comment (jhuleatt) [inline templates/init/functions/typescript/index.ts:14]: nit: this probably doesn't need to be double-commented

File: templates/init/functions/typescript/index.ts
Comment (jhuleatt) [inline templates/init/functions/typescript/index.ts:10]: Need to remove spaces inside curly brackets to make ESLint happy. Also do it in the commented out lines at the top

File: templates/init/functions/javascript/index.js
Comment (taeold) [inline templates/init/functions/javascript/index.js:7]: This is great!

**Review by taeold — APPROVED** [review]

**Review by taeold — APPROVED** [review]

**Pull request merged**
